arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -767,4 +780,95 @@ bool RocksDBHotBackupRestore::createRestoringDirectory(std::string & restoreDirO
 } // RocksDBHotBackupRestore::createRestoringDirectory
 
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief RocksDBHotBackupList
+///        POST:  Returns array of Hotbackup directory names
+////////////////////////////////////////////////////////////////////////////////
+RocksDBHotBackupList::RocksDBHotBackupList(const VPackSlice body)
+  : RocksDBHotBackup(body)
+{
+}
+
+
+RocksDBHotBackupList::~RocksDBHotBackupList() {
+}
+
+
+void RocksDBHotBackupList::parseParameters(rest::RequestType const type) {
+
+  _valid = (rest::RequestType::POST == type);
+
+  if (!_valid) {
+    _result.add(VPackValue(VPackValueType::Object));
+    _result.add("httpMethod", VPackValue("only POST allowed"));
+  } // if
+
+  if (!_valid) {
+    _result.close();
+    _respCode = rest::ResponseCode::BAD;
+    _respError = TRI_ERROR_HTTP_BAD_PARAMETER;
+  } // if
+
+  return;
+
+} // RocksDBHotBackupList::parseParameters
+
+
+// @brief route to independent functions for "create" and "delete"
+void RocksDBHotBackupList::execute() {
+  std::vector<std::string> hotbackups;
+
+  hotbackups = TRI_FilesDirectory(rebuildPathPrefix().c_str());
+
+  // remove working directories from list
+  std::vector<std::string>::iterator found;
+  found = std::find(hotbackups.begin(), hotbackups.end(), dirCreatingString);
+  if (hotbackups.end() != found) {
+    hotbackups.erase(found);
+  }
+
+  found = std::find(hotbackups.begin(), hotbackups.end(), dirRestoringString);
+  if (hotbackups.end() != found) {
+    hotbackups.erase(found);
+  }
+
+  found = std::find(hotbackups.begin(), hotbackups.end(), dirDownloadingString);
+  if (hotbackups.end() != found) {
+    hotbackups.erase(found);
+  }
+
+  // add two failsafe directory name string variables (from different branch)
+  found = std::find(hotbackups.begin(), hotbackups.end(), "FAILSAFE");
+  if (hotbackups.end() != found) {
+    hotbackups.erase(found);
+  }
+
+  found = std::find(hotbackups.begin(), hotbackups.end(), "FAILSAFE.1");
+  if (hotbackups.end() != found) {
+    hotbackups.erase(found);
+  }
+
+
+  try {
+    _result.add(VPackValue(VPackValueType::Object));
+    _result.add("server", VPackValue(getPersistedId()));
+    _result.add("hotbackups", VPackValue(VPackValueType::Array));  // open
+    for (auto dir : hotbackups) {
+      _result.add(VPackValue(dir.c_str()));
+    } // for
+    _result.close();
+    _result.close();
+    _success = true;
+  } catch (...) {
```

Comment:
If this throws, we should bring `_resuit` into a well-defined state.

Reply:
fixed.